actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -264,7 +266,7 @@ def tag_options(options, escape = true)
           sep    = " "
           options.each_pair do |key, value|
             type = TAG_TYPES[key]
-            if type == :data && value.is_a?(Hash)
+            if value.is_a?(Hash) && (type == :data || TagBuilder.nest_html_attributes_within.include?(key))
```

Comment:
Out of curiosity, what would happen if this was enabled for any key, not just the allowed ones?

Reply:
I like that idea! I'd initially considered globally supporting all `Hash` keys, but worried about that change in behavior.

I struggled to imagine what depending on existing behavior might enable, or the benefits to be had. It'd be a "breaking" change in that regard, and is probably worth having a broader discussion.

Reply:
Yeah, I'd rather not make such a big change now.

Reply:
Also it seems `data` and `aria` handle boolean values differently since: https://github.com/rails/rails/pull/40479 . My guess is that `data` is the attribute that is different here.
I'm also not sure if there are a lot of other attributes, that should work similarly to `aria` and/or `data`.